ccl/multiregionccl: TestIndexCleanupAfterAlterFromRegionalByRow failed

[(ccl/multiregionccl).TestIndexCleanupAfterAlterFromRegionalByRow failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2900073&tab=buildLog) on [release-21.1@4e822751a488380f3f39a5fe25ebd529920fefbc](https://github.com/cockroachdb/cockroach/commits/4e822751a488380f3f39a5fe25ebd529920fefbc):

```
=== RUN   TestIndexCleanupAfterAlterFromRegionalByRow
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestIndexCleanupAfterAlterFromRegionalByRow129289207
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestIndexCleanupAfterAlterFromRegionalByRow
    regional_by_row_test.go:678: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestIndexCleanupAfterAlterFromRegionalByRow129289207
--- FAIL: TestIndexCleanupAfterAlterFromRegionalByRow (107.36s)
=== RUN   TestIndexCleanupAfterAlterFromRegionalByRow/GLOBAL
E210420 00:12:18.031014 426800 jobs/adopt.go:260  [n1] 2  job 651451229323132929: adoption completed with error batch timestamp 1618877507.827921694,0 must be after replica GC threshold 1618877536.911364120,0
    regional_by_row_test.go:673: condition failed to evaluate within 45s: expected 4 jobs with status "succeeded", found 3. Jobs found: [{succeeded {"indexes": [{"dropTime": "1618877488983772093", "indexId": 1}], "parentId": 55}} {succeeded {"indexes": [{"dropTime": "1618877492273070720", "indexId": 3}], "parentId": 55}} {succeeded {"indexes": [{"dropTime": "1618877492275367277", "indexId": 4}], "parentId": 55}}]
        goroutine 374373 [running]:
        runtime/debug.Stack(0xc009e29f30, 0x4fbdd40, 0xc004cc9f20)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x5188a60, 0xc0052adc80, 0xc009e29f30)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x6d
        github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestIndexCleanupAfterAlterFromRegionalByRow.func1(0xc0052adc80)
        	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/regional_by_row_test.go:673 +0x5bc
        testing.tRunner(0xc0052adc80, 0xc0079e1690)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
E210420 00:12:18.548271 401574 kv/kvserver/queue.go:1093  [n1,merge,s1,r37/1:/Table/55{-/2/"@"/P…}] 3  failed to send RPC: sending to all replicas failed; last error: r45 was not found on s1
    --- FAIL: TestIndexCleanupAfterAlterFromRegionalByRow/GLOBAL (60.97s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestIndexCleanupAfterAlterFromRegionalByRow PKG=./pkg/ccl/multiregionccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestIndexCleanupAfterAlterFromRegionalByRow.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/multiregionccl/BUILD.bazel
+++ b/pkg/ccl/multiregionccl/BUILD.bazel
@@ -38,6 +38,7 @@ go_test(
         "//pkg/jobs",
         "//pkg/keys",
         "//pkg/kv",
+        "//pkg/kv/kvserver",
         "//pkg/roachpb",
         "//pkg/security",
         "//pkg/security/securitytest",

--- a/pkg/ccl/multiregionccl/regional_by_row_test.go
+++ b/pkg/ccl/multiregionccl/regional_by_row_test.go
@@ -20,6 +20,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/jobs"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv"
@@ -566,7 +567,20 @@ func TestIndexCleanupAfterAlterFromRegionalByRow(t *testing.T) {
 	} {
 		t.Run(tc.locality, func(t *testing.T) {
 			_, sqlDB, cleanup := multiregionccltestutils.TestingCreateMultiRegionCluster(
-				t, 3 /* numServers */, base.TestingKnobs{}, nil, /* baseDir */
+				t, 3 /* numServers */, base.TestingKnobs{
+					Store: &kvserver.StoreTestingKnobs{
+						// Disable the merge queue because it makes this test flakey
+						// under stress. Consider changing this when admin commands are
+						// better synchronized leading to less thrashing of the range cache.
+						// We may also need to retry ClearRange operations which bump into
+						// the GC threshold. Generally that's not a concern because
+						// generally that threshold isn't super small. The problem with
+						// ranges is that when they merge, as may happen during the parallel
+						// execution of a big ClearRange is that the highest threshold will
+						// be inherited.
+						DisableMergeQueue: true,
+					},
+				}, nil, /* baseDir */
 			)
 			defer cleanup()
 

